### PR TITLE
SCRIPTS: Update base elemental magic damage formula

### PR DESCRIPTION
**Pull request by madwraith**
Based on what I read here: https://www.bg-wiki.com/bg/Magic_Damage
My focus is just elemental magic, other types should not be effected by this.
I also had to update all the elemental spells lua for this.
Also fixed many comments in the spell luas.

### CODE REVIEW

**Comment by dazusu** [comment]
Is this going to be merged eventually?

**Comment by teschnei** [comment]
Am I going to be back from Maui eventually?

**Comment by teschnei** [comment]
(Yes, but not until this weekend)

File: scripts/globals/magic.lua [hunk 1]
```diff
@@ -1428,8 +1428,71 @@ function canOverwrite(target, effect, power, mod)
...
+    end		
+
+    --[[
+            Calculate base damage:       
+            D = mDMG + V + (dINT × M)
+            D is then floored
+            For dINT reduce by amount factored into the V value (example: at 134 INT, when using V100 in the calculation, use dINT = 134-100 = 34)             
+        ]]
+    if(dINT <= 0) then 
+        DMG = V + dINT; --unsure how to handle dINT <= 0!!!
```
Comment (teschnei) [inline scripts/globals/magic.lua]: For dINT < 0, reduced M values apply. With a negative dINT, calculated value for D will be lower than V, potentially even 0.

File: scripts/globals/magic.lua [hunk 2]
```diff
@@ -1428,8 +1428,71 @@ function canOverwrite(target, effect, power, mod)
...
+
+    elseif(dINT >= 100 and dINT <= 199) then
+        V = spellParams.V100;
+        M = spellParams.M100;
+        DMG = math.floor(DMG + mDMG + V + ((dINT - 100) * M));
+
+    elseif(dINT > 199) then --same as 100+ till more testing is done!!!
+        V = spellParams.V200;
+        M = spellParams.M200;
+        DMG = math.floor(DMG + mDMG + V + ((dINT - 100) * M));
```
Comment (teschnei) [inline scripts/globals/magic.lua]: just cap it at V200 (there is no M200)

File: scripts/globals/magic.lua [hunk 3]
```diff
@@ -1428,8 +1428,71 @@ function canOverwrite(target, effect, power, mod)
...
-function doElementalNuke(V,M,caster,spell,target,hasMultipleTargetReduction,resistBonus)
-	return doNuke(V,M,caster,spell,target,hasMultipleTargetReduction,resistBonus,ELEMENTAL_MAGIC_SKILL,MOD_INT);
+function doElementalNuke(caster, spell, target, spellParams)
+    local DMG = 0;
+    local V = 0;
+    local M = 0;
+    local dINT = caster:getStat(MOD_INT) - target:getStat(MOD_INT);
+    local hasMultipleTargetReduction = spellParams.hasMultipleTargetReduction; --still unused!!!
+    local resistBonus = spellParams.resistBonus;
+    local mDMG = caster:getStat(MOD_MAGIC_DAMAGE);
```
Comment (teschnei) [inline scripts/globals/magic.lua]: getMod, not getStat

File: scripts/globals/magic.lua [hunk 4]
```diff
@@ -1428,8 +1428,71 @@ function canOverwrite(target, effect, power, mod)
...
+
+    if(mDMG == nil) then
```
Comment (teschnei) [inline scripts/globals/magic.lua]: will never happen if you use the correct call

**Comment by madwraith** [comment]
alright updated everything

**Comment by madwraith** [comment]
handled those couple ga IVs and update all mp cast/recast values in db

**Pull request merged**